### PR TITLE
Gh 47 errors handling

### PR DESCRIPTION
**Pull request by sche**

### CODE REVIEW

File: Multisig/UI/Settings/App Settings/AdvancedAppSettings.swift [hunk 1]
```diff
@@ -25,6 +25,16 @@ struct AdvancedAppSettings: View {
                 KeyValueRow("Transaction service",
                             value: DisplayURL(App.shared.safeTransactionService.url).absoluteString)
             }
+
+            if !(App.configuration.services.environment == .production) {
+                Section(header: SectionHeader("DUBUG")) {
```
Comment (DmitryBespalov) [inline Multisig/UI/Settings/App Settings/AdvancedAppSettings.swift]: Almost "Do Bug" 😉

File: Multisig/Data/Services/Utils/HTTPClient.swift [hunk 2]
```diff
@@ -69,18 +71,22 @@ class HTTPClient {
...
                 } catch {
-                    #warning("TODO: Crashlytics")
+                    LogService.shared.error("Could not decode error details from the data: \(data)")
                     return Message.unexpectedError.rawValue
                 }
-            default:
-                #warning("TODO: Crashlytics")
+            case .unknownError(let request, let response, let data):
+                let requestStr = String(describing: request)
+                let responseStr = response != nil ? String(describing: response!) : ""
```
Comment (DmitryBespalov) [inline Multisig/Data/Services/Utils/HTTPClient.swift]: `let responseStr = response.map { String(describing: $0) } ?? ""`

File: Multisig/Data/Services/Utils/HTTPClient.swift [hunk 3]
```diff
@@ -69,18 +71,22 @@ class HTTPClient {
...
+                let dataStr = data != nil ? String(describing: data!) : ""
```
Comment (DmitryBespalov) [inline Multisig/Data/Services/Utils/HTTPClient.swift]: same here, the Optional has `map` function:

`data.map { String(describing: $0) }  ?? ""`

File: Multisig/Data/Services/Utils/HTTPClient.swift (hunk 2 above)
Comment (DmitryBespalov) [reply Multisig/Data/Services/Utils/HTTPClient.swift]: or even shorter:

`response.map(String.init(describing:)) ?? ""`

**Review by DmitryBespalov — APPROVED** [review]

**Review by MouazAlzahabi — APPROVED** [review]

File: Multisig/UI/Settings/App Settings/AdvancedAppSettings.swift (hunk 1 above)
Comment (sche) [reply Multisig/UI/Settings/App Settings/AdvancedAppSettings.swift]: :)

File: Multisig/Data/Services/Utils/HTTPClient.swift (hunk 2 above)
Comment (sche) [reply Multisig/Data/Services/Utils/HTTPClient.swift]: ok

**Pull request merged**